test/e2e/aks_upgrade.go

```diff
@@ -65,6 +66,7 @@ func AKSUpgradeSpec(ctx context.Context, inputGetter func() AKSUpgradeSpecInput)
 	Eventually(func(g Gomega) {
 		resp, err := managedClustersClient.Get(ctx, infraControlPlane.Spec.ResourceGroupName, infraControlPlane.Name, nil)
 		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(resp.Properties.ProvisioningState).To(Equal(ptr.To("Succeeded")))
```

Comment:
FYI @willie-yao I think this overlaps with a change you added to #4155.

Reply:
Looks like my change is accomplishing the same thing although the provisioning state is coming from the [managed cluster itself](https://github.com/willie-yao/cluster-api-provider-azure/blob/clusterclass-aks-final/test/e2e/aks_upgrade.go#L73) rather than the response properties. Is the behavior any different?